Added a shiny filter and a shiny emoji

Pontiky opened the pull request:
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->


<!--
 Please make all PRs to the dev branch. The dev branch will be
 periodically pulled into the master branch. This allows time for
 the changes and documentation to be tested before being exposed to
 a wider population.
--->

## Description
<!-- In detail, describe what your PR adds to PokeAlarm -->

This PR adds a new filter `can_be_shiny` to filter Pokémons that can be shiny in the wild, in researches or in raids.
A new DTS `shiny_emoji` has been added for pokemon, quest and raid events to show if the Pokemon can be shiny using an emoji.

## Type of Change
<!-- Place a single 'x' into the correct box, ex: [x] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context
<!---
 Why is this change required? What problem does it solve?
 If it fixes an open issue, please link to the issue here.
-->
The `data/shiny_data.json` comes from the [PoGo API](https://pogoapi.net/documentation/) (`shiny_pokemon.json`)
This update was previously requested by someone and will close #746.

## How Has This Been Tested?
<!---
 Please describe in detail how you tested your changes. Make sure to
 describe what tests you have performed, your testing environment,
 and if you have used this in a production setting. Please add
 screenshots if appropriate.
-->

- Tested with the webhook tester on some examples
- Wrote tests cases for each event (mon, quest and raid)
- Tested using a live configuration

## Wiki Update
<!--
 Does this feature require an update to the wiki? If so, please submit
 the required change to https://github.com/RocketMap/PokeAlarmWiki.
 If your feature requires a wiki update, you may submit it for review
 but it will not be accepted until the wiki update is complete.
--->
- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

Included with this PR

Pontiky committed:
added shiny filter and shiny emoji

Pontiky committed:
added shiny doc

Pontiky committed:
added shiny filter unit test

billyjbryant commented:
@Pontiky please rebase this PR

Pontiky commented:
> @Pontiky please rebase this PR

Done

jfberry commented:
Poracle uses its own repository for this as we found this source not reliable
https://github.com/jms412/PkmnShinyMap
I would suggest you consider using this (honest, we don't mind)
(Note poracle checks by id, then id_form and either counts as shiny possible.  The other values in there will be ignored, it's just how jms412 makes his life easier)

billyjbryant commented:
This needs to be rebased

billyjbryant reviewed (changes requested):
Please rebase

Pontiky committed:
change shiny data source and add auto fetch of shiny data

Pontiky commented:
Rebased
- Changed the shiny data source
- Also added the auto fetch of new data like #802, #813 and #823
- Live tests work too!
![image](https://user-images.githubusercontent.com/45545471/161167143-920a8db3-d1da-42ef-b628-dd5c16d0b90a.png)

Pontiky commented:
> Poracle uses its own repository for this as we found this source not reliable https://github.com/jms412/PkmnShinyMap I would suggest you consider using this (honest, we don't mind) (Note poracle checks by id, then id_form and either counts as shiny possible. The other values in there will be ignored, it's just how jms412 makes his life easier)

Just for traceback purposes, after some discussions with you (jfberry), Jms412 (the host of this mentioned shiny source) and the PA squad, [we decided to use this new file to fetch shiny possible mons](https://discord.com/channels/215181169761714177/229088248146821140/951431871114399764).

I made the necessary changes with my last rebase and commit.

jfberry commented on start_pokealarm.py:
```diff
@@ -97,6 +97,8 @@ def check_for_update():
     masterfile_vreq = "https://api.github.com/repos/WatWowMap/Masterfile-" \
         "Generator/commits?path=master-latest-everything.json&per_page=1"
     pogoapi_vreq = "https://pogoapi.net/api/v1/api_hashes.json"
+    shiny_possible_vreq = "https://api.github.com/repos/jms412/PkmnShinyMap/" \
```
I think this url might be better:
`https://raw.githubusercontent.com/jms412/PkmnShinyMap/main/shinyPossible.json`

Pontiky replied:
This first url is to make a request to the github api to obtain the last commit signature and know if PA need to update the local file.

jfberry commented on start_pokealarm.py:
```diff
@@ -285,6 +296,42 @@ def download_data(sigdiff=None):
+        log.info("New shiny_possible data found! Fetching in progress...")
+
+        # Fetch data
+        shiny_possible_url = "https://raw.githubusercontent.com/jms412/" \
```
but you have the raw url here, so perhaps I misunderstood the other one

jfberry commented on start_pokealarm.py:
```diff
@@ -285,6 +296,42 @@ def download_data(sigdiff=None):
+        shiny_possible_data = requests.get(shiny_possible_url).json()
+
+        # Check some dict paths which don't have to change
+        if shiny_possible_data["map"]["1"] != " \u2728":
```
Poracle actually only cares that it is present and non-empty. In the old days the map was used to give us the symbol but no longer.  I am sure this will be preserved so just an FYI.

Pontiky replied:
That's just some security checks we use to do for each remotely file ^^

LegitDongo commented on PokeAlarm/Utilities/MonUtils.py:
```diff
@@ -147,3 +147,27 @@ def get_pokemon_gender(gender):
+
+
+# Returns shiny symbol if the pokemon is shiny in the wild
+def get_wild_shiny_emoji(pokemon_id, form_id):
```
This isn't used anywhere, looks like it's duplicated from Utils

LegitDongo reviewed (changes requested):
Haven't tested, a quick look-through looks good

Pontiky committed:
removed unnecessary function

billyjbryant reviewed (approved):
Read through the changes, also saw where this was tested. I see no problem approving for merger.

LegitDongo reviewed (approved):
LGTM
